src/ConnectedMode/Binding/UnintrusiveBindingPathProvider.cs

```diff
@@ -38,35 +35,28 @@ internal interface IUnintrusiveBindingPathProvider
     [Export(typeof(IUnintrusiveBindingPathProvider))]
     internal class UnintrusiveBindingPathProvider : IUnintrusiveBindingPathProvider
     {
-        private readonly IVsSolution solution;
+        private readonly ISolutionInfoProvider solutionInfoProvider;
 
         private readonly string SLVSRootBindingFolder;
 
         [ImportingConstructor]
-        public UnintrusiveBindingPathProvider([Import(typeof(SVsServiceProvider))]IServiceProvider serviceProvider,
-            IThreadHandling threadHandling)
-            : this(serviceProvider, threadHandling, EnvironmentVariableProvider.Instance)
+        public UnintrusiveBindingPathProvider(ISolutionInfoProvider solutionInfoProvider)
+            : this(solutionInfoProvider, EnvironmentVariableProvider.Instance)
         {
         }
 
-        internal /* for testing */ UnintrusiveBindingPathProvider(IServiceProvider serviceProvider,
-            IThreadHandling threadHandling,
+        internal /* for testing */ UnintrusiveBindingPathProvider(ISolutionInfoProvider solutionInfoProvider,
             IEnvironmentVariableProvider environmentVariables)
         {
             SLVSRootBindingFolder = Path.Combine(environmentVariables.GetSLVSAppDataRootPath(), "Bindings");
-
-            IVsSolution slnService = null;
-            threadHandling.RunOnUIThread(() => slnService = serviceProvider.GetService(typeof(SVsSolution)) as IVsSolution);
```

Comment:
We shouldn't be switching threads in a MEF constructor.
The new version avoids this.